### PR TITLE
dynamic plugins support custom paths

### LINKED ISSUE #1703: The dynamic plugin installation should be able to customize the plugin path.
### What would you like to be enhanced?

The dynamic plugin installation should be able to customize the plugin path.

### Why is this needed?

The dynamic plugin installation should be able to customize the plugin path.

### PR DESCRIPTION
**Pull request by daizhenyu**
**What type of PR is this?**

Refactor.

**What this PR does / why we need it?**

dynamic plugins support custom paths

**Which issue(s) this PR fixes？**

Fixes #1703

**Does this PR introduce a user-facing change?**

No

## Checklist
- [ ] Make sure there is a [GitHub_issue](https://github.com/sermant-io/Sermant/issues) related with this PR before you start working on it.
- [ ] Make sure you have squashed your change to one single commit.
- [ ] GitHub Actions works fine in this PR.

### CODE REVIEW

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/command/DynamicAgentArgsManager.java [hunk 1]
```diff
@@ -31,6 +31,8 @@ public class DynamicAgentArgsManager {
      */
     private static final Map<String, String> AGENT_ARGS = new HashMap<>();
 
+    private static final String DYNAMIC_PLUGIN_PATH_KEY = "sermant_dynamic_plugins_path";
```
Comment (lilai23) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/command/DynamicAgentArgsManager.java]: SERMANT_PLUGIN_PACKAGE_PATH

Comment (daizhenyu) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/command/DynamicAgentArgsManager.java]: done

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/plugin/PluginManager.java [hunk 2]
```diff
@@ -414,4 +420,20 @@ private interface JarFileConsumer {
...
+        return BootArgsIndexer.getPluginPackageDir().getCanonicalPath();
+    }
+
+    private static String getDynamicPluginPackagePath() throws IOException {
+        String dynamicPluginPath = DynamicAgentArgsManager.getDynamicPluginPackagePath();
+        if (!StringUtils.isEmpty(dynamicPluginPath)) {
+            return dynamicPluginPath;
+        }
+        String agentPath = DynamicAgentArgsManager.getAgentPath();
+        if (!StringUtils.isEmpty(dynamicPluginPath)) {
```
Comment (zwmagic) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/plugin/PluginManager.java]: judge whether agentPath is empty

Comment (daizhenyu) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/plugin/PluginManager.java]: done

**Review by zwmagic — DISMISSED** [review]

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/command/DynamicAgentArgsManager.java [hunk 3]
```diff
@@ -52,4 +56,22 @@ public static void refreshAgentArgs(Map<String, String> newAgentArgs) {
...
         return AGENT_ARGS.get(key);
     }
+
+    /**
+     * get dynamic plugin path
+     *
+     * @return dynamic plugin path
+     */
+    public static String getDynamicPluginPackagePath() {
+        return AGENT_ARGS.get(SERMANT_PLUGIN_PACKAGE_PATH_KEY);
```
Comment (lilai23) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/command/DynamicAgentArgsManager.java]: decide to use (CommonConstant.AGENT_FILE_KEY)

Comment (daizhenyu) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/command/DynamicAgentArgsManager.java]: done

**Review by lilai23 — APPROVED** [review]

**Review by hanbingleixue — APPROVED** [review]

**Pull request merged**